site/components/map/ServicesMap.tsx

```diff
@@ -280,30 +316,21 @@ const Map = ({
         } else {
             if (showSelectAllText) {
                 setLoading(true);
-                const atcoCodes = includeMarkerData ? markerData.map((marker) => marker.atcoCode).splice(0, 100) : [];
-
-                const servicesInPolygon = includeMarkerData
-                    ? dataSource
-                        ? await fetchServicesByStops({ atcoCodes, includeRoutes: true, dataSource: dataSource })
-                        : await fetchServicesByStops({ atcoCodes, includeRoutes: true })
+                const atcoCodes = getAtcoCodesFromSelectedStops(markerData);
+                const servicesInPolygon = !!markerData
+                    ? await fetchServicesByStops({ atcoCodes, includeRoutes: true, dataSource: dataSource })
                     : [];
 
                 const servicesStopsInPolygon = servicesInPolygon.flatMap((service) => service.stops);
 
-                const markerDataInAService = includeMarkerData
+                const markerDataInAService = !!markerData
                     ? getMarkerDataInAService(markerData, servicesStopsInPolygon, servicesInPolygon)
                     : [];
 
-                const servicesToAdd = servicesInPolygon
-                    .filter((service) =>
-                        service.stops.filter((stop) => markerData.map((marker) => marker.atcoCode).includes(stop)),
-                    )
-                    .map((service) => serviceSchema.parse(service));
-
                 setSelectedServices(
                     [
                         ...(selectedServices ?? []),
-                        ...(includeMarkerData
+                        ...(!!markerData
```

Comment:
if markerData is an empty array this will still run do we want a length check too?

Reply:
added length check